### PR TITLE
Courses test

### PR DESCRIPTION
**Pull request by Taiwopeter-babs**
- add tests for course
- update delete status code
- update successful delete response to return empty object `{}`

### CODE REVIEW

File: database/migrations/20231109103527_users_courses_table.js
Comment (VivianDee) [inline database/migrations/20231109103527_users_courses_table.js:1]: I was not able to run this migration on the server. I got the error: Only one column can be incremented in a table. That's why it was changed to bigInteger

Comment (Taiwopeter-babs) [reply database/migrations/20231109103527_users_courses_table.js:1]: Has the node version on the server been updated?

Comment (VivianDee) [reply database/migrations/20231109103527_users_courses_table.js:1]: Not yet.

Comment (Taiwopeter-babs) [reply database/migrations/20231109103527_users_courses_table.js:1]: Maybe it's my mysql version (8.0.2). Using `bigInteger` causes errors. what version is the mysql on the server?

Comment (Taiwopeter-babs) [reply database/migrations/20231109103527_users_courses_table.js:1]: @VivianDee  The users_courses table issue has been corrected,

**Review by Mistaweng — APPROVED** [review]

**Comment by Mistaweng** [comment]
Great Job on the database

**Pull request merged**